README.md

```diff
@@ -1,2 +1,104 @@
-# styleguide
-contains runnable coding guidelines to maintain consistency across all our repos  
+# Runnable Code and Test Patterns Guide
+
+Runnable's approach to writing code and tests.
+
+## Table of Contents
+
+1. [Code Formatting](#code-formatting)
+1. [Testing](#testing)
+    1. [Types of Tests](#types-of-tests)
+    1. [Stubbing](#stubbing)
+    1. [Asserting](#asserting)
+    1. [Additional Testing Reading](#additional-testing-reading)
+
+## Code Formatting
+
+[![js-standard-style](https://cdn.rawgit.com/feross/standard/master/badge.svg)](https://github.com/feross/standard)
+
+We use [`standard`](http://standardjs.com/) in most projects. If it passes `standard`, anything else is up to the developer. No questions asked.
+
+**[back to the top](#table-of-contents)**
+
+## Testing
+
+Let's discuss various tests, what they should do and cover, and how best to organize them.
+
+### Types of Tests
+
+There are three types of tests we should be concerned about.
+
+- Unit
+- Integration
+- Functional
+
+#### Definitions
+
+**Unit** tests are pretty obvious (or should be). They test exactly one unit of functionality. Most likely this should follow the boundaries of functions or at largest, modules. If we're finding we're writing a lot unit tests for a specific function, it may need to be broken up.
+
+An important note: unit tests should _never_ touch an external service (e.g. database).
```

Comment:
Yeah, I guess I didn't answer the second half...

Yes, those tests should eventually be fixed. However, this is a document saying how things _should_ be, not what our goals are to address any issues we may have in our current tests. As far as I'm concerned, as people come in contact with sub-par tests as they are adding functionality or modifying them, any tests they touch should be updated accordingly. No one developer is going to have to (or want to) go through a 30-function-module and write tests for each one, but if one at a time is addressed, things will improve.